src/graph/graph.c

```diff
@@ -967,28 +933,25 @@ void _BulkDeleteEdges(Graph *g, Edge *edges, size_t edge_count) {
 		GrB_Descriptor_set(desc, GrB_OUTP, GrB_REPLACE);
 
 		for(int r = 0; r < relationCount; r++) {
+			GrB_Matrix R;
 			GrB_Matrix mask = masks[r];
-			GrB_Matrix R = Graph_GetRelationMatrix(g, r); // Relation Matrix.
 			if(mask) {
-				GrB_Matrix M = Graph_GetRelationMap(g, r);  // Relation mapping matrix.
-				// Remove every entry of R and M marked by Mask.
+				R = Graph_GetRelationMatrix(g, r);  // Relation matrix.
+				// Remove every entry of R marked by Mask.
 				// Desc: GrB_MASK = GrB_SCMP,  GrB_OUTP = GrB_REPLACE.
 				// R = R & !mask.
-				GrB_Matrix_apply(R, mask, GrB_NULL, GrB_IDENTITY_BOOL, R, desc);
-				// M = M & !mask.
-				GrB_Matrix_apply(M, mask, GrB_NULL, GrB_IDENTITY_UINT64, M, desc);
+				GrB_Matrix_apply(R, mask, GrB_NULL, GrB_IDENTITY_UINT64, R, desc);
+
+				// Collect remaining edges. remaining_mask = remaining_mask + R.
+				GrB_eWiseAdd_Matrix_Semiring(remaining_mask, GrB_NULL, GrB_NULL, GxB_ANY_PAIR_BOOL, remaining_mask,
+											 mask, GrB_NULL);
+
```

Comment:
This needs to be outside the `if(mask)`. You need to collect the remaining edges regardless to mask

Reply:
👍